wavedump/src/fit-histograms

```diff
@@ -0,0 +1,566 @@
+                
+                f1.FixParameter(0, scale)
+                # f1.SetParameter(0, scale)
+                # f1.SetParLimits(0, scale*0.1, scale * 100)
+                
+                f1.FixParameter(1, offset)
+                
+                f1.SetParameter(2, l)
+                
+                f1.SetParameter(3, SPE_charge)
+                f1.SetParLimits(3, zero_peak_end - offset, SPE_charge + 1)
+                
+                f1.FixParameter(4, raw_spread)
+                
+                f1.FixParameter(5, D_SPE_CHARGE_SPREAD)
+                
+                # The following indented block is code older than June 2022
+                    # if len(x_pos) >= 1:
+                    #     if len(x_pos) == 1:
+                    #         f1.FixParameter(1, float(x_pos))  
+                    #         f1.SetRange(f1.GetParameter(1) - h.GetStdDev() * 0.5, f1.GetXmax()) 
+                    #     else:
+                    #         f1.FixParameter(1, float(x_pos[0]))  
+                    #         f1.SetRange(f1.GetParameter(1) - h.GetStdDev() * 0.5, float(x_pos[-1]) + h.GetStdDev() * 0.5) 
+                    # if len(x_pos) > 1:
+                    #     found_SPE_peak = True
+                    #     f1.FixParameter(3, float(x_pos[1]) - f1.GetParameter(1))
+                    # if not found_SPE_peak:
+                    #     f1.SetParameter(3,0.8)                
+                    #     f1.SetParLimits(3,0.1,5.)
+                    # if len(x_pos) == 0:
+                    #     f1.SetParameter(1, -0.08)
+                    #     f1.SetParLimits(1,-1,1)
+
+                    #for j, i in enumerate(range(0,n*3,3)):
+                    #    print(i)
+                    #    f1.SetParameter(i,y_pos[j])
+                    #    f1.SetParameter(i+1,x_pos[j])
+                    #    f1.FixParameter(i+1,x_pos[j])
+                    #    f1.SetParameter(i+2,0.02)
+                    #    f1.FixParameter(i+2,0.02)
+
+                    #r = h.Fit(f1, 'ILMSRN+')
+                
+                for i in range(6):
+                    print(f'[{i}]: {f1.GetParameter(i)}')
+                
+                r = h.Fit(f1, 'SRB+')
+
+                for i in range(7):
+                    f1.ReleaseParameter(i) 
+                f1.FixParameter(1, f1.GetParameter(1)) 
+                f1.FixParameter(4, f1.GetParameter(4)) 
+                # f1.SetParLimits(1, f1.GetParameter(1) - 0.1, f1.GetParameter(1) + 0.1) 
+                f1.SetParLimits(2, max(0, f1.GetParameter(2) - 1), f1.GetParameter(2) + 1) 
+                f1.SetParLimits(3, max(zero_peak_end-offset, f1.GetParameter(3) - 1), f1.GetParameter(3) + 1) 
+                # f1.SetParLimits(4, 0, f1.GetParameter(4) + 0.5) 
+                f1.SetParLimits(5, 0, f1.GetParameter(5) + 0.1) 
+                f1.SetParLimits(6, 0, 0.25) 
+                
+                f1.SetLineColor(3)
+                r = h.Fit(f1, 'SR+')
```

Comment:
I would also add a comment above about the general strategy, i.e. explain the order in which we do the fits, how we estimate the parameters one by one.